### PR TITLE
[Feature] Search for Stories

### PR DESCRIPTION
**Pull request by FluentCoding**
You can now filter the story panel for stories of specific people and hide the other ones. It works case-insensitive and updates the entries everytime the searchbox value changes (currently synchronously).

Bildschirmfoto von 2020-10-31 22-08-55

### CODE REVIEW

**Comment by FluentCoding** [comment]
The only problem is that it only searches for stories that are already fetched by the client. What about adding a new API endpoint for searching all available stories?

**Comment by benawad** [comment]
do you want to add the backend part or would you like me to?

**Comment by FluentCoding** [comment]
I could give it a try tomorrow, but you would have to explain me then how to set it up on my local machine (proxy?) which would btw help me long-term contributing to this project, otherwise you can just implement it for yourself.

**Comment by PranjalAgni** [comment]
@FluentCoding Let me know, I would like to work on this

**Comment by FluentCoding** [comment]
> @FluentCoding Let me know, I would like to work on this

@PranjalAgni Yeah, as I've said, the only problem is that I don't know how to set up benawad's backend server, and that's the only problem (I barely ever touched Node.js and things like Express.js before) for me to work on it. It would be pretty cool, if someone could tell me how to do that, but you can ofc work on the feature no matter whether I'm able to configure the backend server or not.

**Comment by benawad** [comment]
it'll probably be hard to do if you've never done node/express/postgresql before, I'll add it or @PranjalAgni

**Comment by FluentCoding** [comment]
> it'll probably be hard to do if you've never done node/express/postgresql before, I'll add it or @PranjalAgni

My expertise is in software engineering. And I'm familiar with SQL and NoSQL database systems so it actually shouldn't be a problem. Also, I have a react project (not angular dw lol) rn so I'm learning quite fast, but as I've said, I have no clue how you set up your local dev env for your backend server.

**Comment by PranjalAgni** [comment]
@FluentCoding As you will be working on the frontend part of this, we can divide the work what's say? 
 
Also for the backend dev environment I have set it up inside a docker container but for ease you can install PostgresSQL + PGAdmin and we are using Koa to build our rest api it's similar to express

**Comment by FluentCoding** [comment]
> @FluentCoding As you will be working on the frontend part of this, we can divide the work what's say? 
>  
> For the backend dev environment I have set it up inside a docker container but for ease you can install PostgresSQL + PGAdmin and we are using Koa to build our rest api it's similar to express 
> 

Alright, we can probably do that. I will message you, if it won't work (school, ...) but I guess that we can start tomorrow (I have almost 5 AM rn lol).

**Comment by PranjalAgni** [comment]
> > @FluentCoding As you will be working on the frontend part of this, we can divide the work what's say?
> > For the backend dev environment I have set it up inside a docker container but for ease you can install PostgresSQL + PGAdmin and we are using Koa to build our rest api it's similar to express
> 
> Alright, we can probably do that. I will message you, if it won't work (school, ...) but I guess that we can start tomorrow (I have almost 5 AM rn lol).

Yes let me know if you get stuck somewhere while setting up backend workspace. Happy to help

**Comment by FluentCoding** [comment]
@PranjalAgni Alright, I've completed it kinda. (I hope its ok for you that I did it I just had nothing to do and I'm quite addicted to programming lol). Do you want to try it out tho? I'm not sure if it works flawlessly.

**Comment by PranjalAgni** [comment]
> @PranjalAgni Alright, I've completed it kinda. (I hope its ok for you that I did it I just had nothing to do and I'm quite addicted to programming lol). Do you want to try it out tho? I'm not sure if it works flawlessly.

No problem I was also going to work on it but you have raised PR now so I may have to drop it

**Comment by bar9** [comment]
@FluentCoding @PranjalAgni can you summarize the state of this PR? Are you two still collaborating on this?

**Comment by FluentCoding** [comment]
> @FluentCoding @PranjalAgni can you summarize the state of this PR? Are you two still collaborating on this?

I've already completed the implementation, I'm basically just waiting for someone to extensively test and merge it (benawad, but I've heard that he isn't maintaining this project anymore).

**Comment by TheFern2** [comment]
@bar9 this PR was done pre svelte, it will need to be redone.
@FluentCoding I'd take a look about adding a search functionality in the api in the meantime.

**Comment by FluentCoding** [comment]
> @bar9 this PR was done pre svelte, it will need to be redone.
> @FluentCoding I'd take a look about adding a search functionality in the api in the meantime.
> 
> Edit: Nvm you also had submitted a search api PR.

Yep.
